packages/hint-validate-summary/src/hint.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @fileoverview check for summary tag and look for the computedStyle to check whether it is the same as it should be
+ */
+
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint } from 'hint/dist/src/lib/types';
+import { StyleEvents, StyleParse } from '@hint/parser-css';
+import meta from './meta';
+const selectorParser = require('postcss-selector-parser');
+
+
+export default class ValidateSummaryHint implements IHint {
+    public static readonly meta = meta;
+    public selectorNames: string[] = [];
+
+    public constructor(context: HintContext<StyleEvents>) {
+
+
+        const validateStyle = ({ ast, resource }: StyleParse) => {
+            const requiredAttribute = 'display';
+            const requiredValue = 'list-item';
+            const parser = selectorParser();
+
+
+            ast.walkRules((rule) => {
+
+                if (
+                    rule.selector.indexOf('summary') > -1
+                ) {
+
+                    const validSelectors = parser.astSync(rule.selector).nodes.filter((node: { nodes: any }) => { // eslint-disable-line no-sync
+                        const { nodes } = node;
+                        let foundCombinator = false;
+
+                        for (let i = nodes.length - 1; i >= 0; i--) {
+                            const element = nodes[i];
+
+                            if (element.type === 'combinator') {
+                                foundCombinator = true;
+                                break;
+                            }
+
+                            if (element.type === 'tag' && element.value === 'summary') {
+                                break;
+                            }
+                        }
+
+                        if (foundCombinator) {
+                            return false;
+                        }
+
+                        return true;
```

Comment:
I think it's possible to get here without encountering a `tag` of `summary`. By changing the above conditions to return, we'll be guaranteed not to get here unless it's that case, so we should return `false`.
```suggestion
                        return false;
```